### PR TITLE
Implement a few trig ops

### PR DESCRIPTION
**Pull request by willtebbutt**

### CODE REVIEW

**Comment by tpgillam** [comment]
Shall we add hyperbolic trig funcs too whilst about it? And inverses (apart from `atan2` since that needs to be done slightly differently because it has two arguments)

**Comment by willtebbutt** [comment]
@tpgillam ready for review

File: test/ops/simple.jl [hunk 1]
```diff
@@ -1,5 +1,11 @@
+_constrained_input_data = Block([
```
Comment (tpgillam) [inline test/ops/simple.jl:1]: maybe `_b_constrained`? Just since elsewhere in the tests we're using the `b_` prefix to indicate something of type `Block`.

**Comment by tpgillam** [comment]
LGTM!

**Pull request merged**